Add string and byte serializer

thmshmm opened the pull request:
fixes #27 

Added support to serialize string and byte array keys/values. Also fixes producing Protobuf data when the data is converted to a byte array upfront.

mostafa commented on serde.go:
```diff
@@ -0,0 +1,43 @@
+	case "io.confluent.kafka.serializers.KafkaAvroSerializer":
+		return SerializeAvro
+	default:
+		return SerializeAvro
```
I think the default case should be a string.

thmshmm replied:
I can change it to string, just thought to keep it compatible with some already existing tests which expect the previous behavior. Shall I still replace it?

thmshmm replied:
If e.g. Protobuf schemas will be supported in the future as well, then we should make it somehow required to specify the serializer not just providing a schema. Maybe the `produce` function should be dedicated to string serialization and the schema options should be removed. If a schema is used it should be provided using the `produceWithConfiguration` together with the appropriate serializer/deserializer configuration.

mostafa replied:
The `produceWithConfiguration` isn't exported to JS because it's not an exported function in Go. Still, having a single uniform interface, `produce`, would be better if we could have optional arguments so that we don't pass an empty string for the schema (and possibly no nulls for others).
I made an initial proposal and built it on top of @iamelevich's work (https://github.com/mostafa/xk6-kafka/pull/20) to pass an object to the writer and the reader interface for configuration, but it didn't turn out as expected. I also included many other options which were left out in the writer configuration in the struct but to no avail.

mostafa reviewed (changes requested):
Overall LGTM, but needs a small change. I have a question: might a user need to receive a byte array while consuming messages?

thmshmm commented:
Yes, deserialization should be implemented as well. I can work on it if I have some time. For now the regular consumer can be used to "just" read some messages and check if they were written.

mostafa commented:
@thmshmm Let's wait until you add that part, too, before merging the PR.

thmshmm commented:
@mostafa can you review again pls? Added the deserialisation part as well.

mostafa commented:
Hey @thmshmm,

Thanks again for your contribution. Have you tested this code? I tried it, and it fails on a check:
```bash
✗ payload is correct
↳  0% — ✓ 0 / ✗ 7
```
And the reason is that the returned data is a string, not a byte array.

thmshmm commented:
It works for me `✓ payload is correct`. I also noticed this, but found no solution. Thats why I used `String.fromCharCode(...msgs[0].value.split(","))` to convert it back. I think this behaviour may come from the goja implementation that k6 uses.

mostafa commented:
I used your script called `scripts/test_bytes.js`, which doesn't work. I will try again to see if I can find a solution for it.

thmshmm commented:
Do you have any errors when creating the k6 bin?
`xk6 build --with github.com/thmshmm/xk6-kafka@1fd4880ca75cdf1c78930e419d192e3bf36397b82 --replace github.com/thmshmm/xk6-kafka@latest=github.com/mostafa/xk6-kafka@latest --output k6`

mostafa commented:
@thmshmm Just found the problem! I'll make suggestions, so you can merge them right away into your PR.

mostafa commented on scripts/test_bytes.js:
```diff
@@ -0,0 +1,69 @@
+*/
+
+import { check } from "k6";
+import { writer, produceWithConfiguration, reader, consume, createTopic } from "k6/x/kafka"; // import kafka extension
```
```suggestion
import {
    writer,
    produceWithConfiguration,
    reader,
    consumeWithConfiguration,
    createTopic,
} from "k6/x/kafka"; // import kafka extension
```

mostafa commented on scripts/test_bytes.js:
```diff
@@ -0,0 +1,69 @@
+    check(messages, {
+        "10 messages returned": (msgs) => msgs.length == 10,
+        "key starts with 'test-id-' string": (msgs) => msgs[0].key.startsWith("test-id-"),
+        "payload is correct": (msgs) => String.fromCharCode(...msgs[0].value.split(",")) === payload,
```
```suggestion
        "payload is correct": (msgs) => String.fromCharCode(...msgs[0].value) === payload,
```

mostafa reviewed (changes requested):
These are the changes you need to make to make the script work as expected. As far as I see, the implementation works as expected, but the script should be fixed.

thmshmm commented:
@mostafa thanks for your support, I integrated all your corrections. But, for me line 62 only works like this
`"payload is correct": (msgs) => String.fromCharCode(...msgs[0].value.split(",")) === payload,` with the `.split(",")`. Do you have any idea?

mostafa reviewed (approved):
@thmshmm Thanks again for your contribution! I'll merge this and release a version after that.

As for your question, I tried these after `let messages = consumeWithConfiguration(consumer, 10, configuration);`:
```javascript
console.log(messages[0].value);
console.log(typeof messages[0].value);
console.log(typeof messages[0].value[0]);
```
and this is the k6 console output:
```bash
INFO[0000] 98,121,116,101,32,97,114,114,97,121,32,112,97,121,108,111,97,100  source=console
INFO[0000] object                                        source=console
INFO[0000] number                                        source=console
```
I also tried [String.fromCharCode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode), and it only accepts a sequence (array?) of numbers and returns a `String`. Anything else isn't supported and will result in an empty string.
